### PR TITLE
[fix] transmission: allow from_task with multiple servers

### LINKED ISSUE #3331: Transmission cann't handle two clients when crossmatch
### Expected behaviour:

Client1 + Client2 both purge seeds.

### Actual behaviour:
Client1 purged, but client2 doesn't.
Because teansmission-rpc will remeber the last client.

### Steps to reproduce:
use from_transmission_client2 crossmatch from_transmission_client1(which produce title that want to be purged)

#### Config:
```
  clean.purge_reseed_seeds:
    free_space:
      abort_if: above
      space: 30000
    <<: *from_transmission_reseed #Client2
    crossmatch:
      from: [from_task: purge.finished_seeds]
      fields: [title]
      action: accept
    <<: *purge_transmission_reseed

  purge.finished_seeds:
    limit_new: 2
    <<: *from_transmission #Client1
    imdb_lookup: yes
    sort_by: transmission_date_done
    if:
      - transmission_progress != 100.0: reject
    list_match:
      from:
        - movie_list: symlinked-movies
    <<: *purge_transmission
```
  
#### Log:
<details>
 <summary>(click to expand)</summary>

```
2022-01-28 21:14:26 DEBUG    task          clean.purge_reseed_seeds executing clean.purge_reseed_seeds
2022-01-28 21:14:26 DEBUG    template      clean.purge_reseed_seeds Merging template global into task clean.purge_reseed_seeds
2022-01-28 21:14:26 DEBUG    log_filter    clean.purge_reseed_seeds Log filter added (config: [{'plugin': 'free_space', 'task': 'clean.purge_reseed_seeds'}])
2022-01-28 21:14:26 DEBUG    disable       clean.purge_reseed_seeds Disabled plugins: status, history, seen_info_hash, seen
2022-01-28 21:14:26 DEBUG    transmission  clean.purge_reseed_seeds Connecting to http://192.168.1.102:9092/transmission/
2022-01-28 21:14:26 DEBUG    backlog       clean.purge_reseed_seeds 0 entries purged from backlog
2022-01-28 21:14:26 VERBOSE  details       clean.purge_reseed_seeds Produced 10 entries.
2022-01-28 21:14:26 VERBOSE  from_task     clean.purge_reseed_seeds Running task `purge.finished_seeds` as subtask.
2022-01-28 21:14:26 DEBUG    task          clean.purge_reseed_seeds>purge.finished_seeds executing clean.purge_reseed_seeds>purge.finished_seeds
2022-01-28 21:14:26 DEBUG    template      clean.purge_reseed_seeds>purge.finished_seeds Merging template global into task clean.purge_reseed_seeds>purge.finished_seeds
2022-01-28 21:14:27 DEBUG    log_filter    clean.purge_reseed_seeds>purge.finished_seeds Log filter added (config: [{'plugin': 'free_space', 'task': 'clean.purge_reseed_seeds>purge.finished_seeds'}])
2022-01-28 21:14:27 DEBUG    disable       clean.purge_reseed_seeds>purge.finished_seeds Disabled plugins: history, seen_info_hash, seen
2022-01-28 21:14:27 DEBUG    remember_rej  clean.purge_reseed_seeds>purge.finished_seeds Task config has changed since last run, purging remembered entries.
2022-01-28 21:14:27 DEBUG    transmission  clean.purge_reseed_seeds>purge.finished_seeds Connecting to http://192.168.1.101:9091/transmission/
2022-01-28 21:14:27 DEBUG    backlog       clean.purge_reseed_seeds>purge.finished_seeds 0 entries purged from backlog
2022-01-28 21:14:27 VERBOSE  details       clean.purge_reseed_seeds>purge.finished_seeds Produced 3 entries.
2022-01-28 21:14:27 DEBUG    manipulate    clean.purge_reseed_seeds>purge.finished_seeds field: `title` from_field: `title` field_value: `sprd-790 Kaori Otosaki.mp4`
2022-01-28 21:14:27 DEBUG    manipulate    clean.purge_reseed_seeds>purge.finished_seeds field: `title` from_field: `title` field_value: `sprd-790 Kaori Otosaki.mp4`
2022-01-28 21:14:27 DEBUG    manipulate    clean.purge_reseed_seeds>purge.finished_seeds field `title` after replace: `sprd-790 Kaori Otosaki.mp4`
2022-01-28 21:14:27 DEBUG    manipulate    clean.purge_reseed_seeds>purge.finished_seeds field: `title` from_field: `title` field_value: `Reset.S01.2022.2160p.WEB-DL.H265.DDP-TJUPT`
2022-01-28 21:14:27 DEBUG    manipulate    clean.purge_reseed_seeds>purge.finished_seeds field: `title` from_field: `title` field_value: `Reset.S01.2022.2160p.WEB-DL.H265.DDP-TJUPT`
2022-01-28 21:14:27 DEBUG    manipulate    clean.purge_reseed_seeds>purge.finished_seeds field `title` after replace: `Reset.S01.2022.2160p.WEB-DL.H265.DDP-TJUPT`
2022-01-28 21:14:27 DEBUG    manipulate    clean.purge_reseed_seeds>purge.finished_seeds field: `title` from_field: `title` field_value: `Eternals.2021.BluRay.1080p.DTS-HD.MA5.1.x265.10bit-BeiTai`
2022-01-28 21:14:27 DEBUG    manipulate    clean.purge_reseed_seeds>purge.finished_seeds field: `title` from_field: `title` field_value: `Eternals.2021.BluRay.1080p.DTS-HD.MA5.1.x265.10bit-BeiTai`
2022-01-28 21:14:27 DEBUG    manipulate    clean.purge_reseed_seeds>purge.finished_seeds field `title` after replace: `Eternals.2021.BluRay.1080p.DTS-HD.MA5.1.x265.10bit-BeiTai`
2022-01-28 21:14:27 VERBOSE  manipulate    clean.purge_reseed_seeds>purge.finished_seeds Modified 0 entries.
2022-01-28 21:14:27 DEBUG    sort_by       clean.purge_reseed_seeds>purge.finished_seeds sorting entries by: transmission_date_done
2022-01-28 21:14:27 DEBUG    if            clean.purge_reseed_seeds>purge.finished_seeds sprd-790 Kaori Otosaki.mp4 matched requirement now - transmission_date_done < timedelta(days=2)
2022-01-28 21:14:27 VERBOSE  task          clean.purge_reseed_seeds>purge.finished_seeds REJECTED: `sprd-790 Kaori Otosaki.mp4` by if plugin because matched requirement: now - transmission_date_done < timedelta(days=2)
2022-01-28 21:14:27 DEBUG    if            clean.purge_reseed_seeds>purge.finished_seeds Reset.S01.2022.2160p.WEB-DL.H265.DDP-TJUPT matched requirement now - transmission_date_done < timedelta(days=2)
2022-01-28 21:14:27 VERBOSE  task          clean.purge_reseed_seeds>purge.finished_seeds REJECTED: `Reset.S01.2022.2160p.WEB-DL.H265.DDP-TJUPT` by if plugin because matched requirement: now - transmission_date_done < timedelta(days=2)
2022-01-28 21:14:27 DEBUG    if            clean.purge_reseed_seeds>purge.finished_seeds Eternals.2021.BluRay.1080p.DTS-HD.MA5.1.x265.10bit-BeiTai matched requirement transmission_status == 'stopped'
2022-01-28 21:14:27 VERBOSE  task          clean.purge_reseed_seeds>purge.finished_seeds ACCEPTED: `Eternals.2021.BluRay.1080p.DTS-HD.MA5.1.x265.10bit-BeiTai` by if plugin because matched requirement: transmission_status == 'stopped'
2022-01-28 21:14:27 DEBUG    imdb_lookup   clean.purge_reseed_seeds>purge.finished_seeds lookup for Eternals.2021.BluRay.1080p.DTS-HD.MA5.1.x265.10bit-BeiTai
2022-01-28 21:14:27 DEBUG    imdb.db       clean.purge_reseed_seeds>purge.finished_seeds movie `Eternals` age 1 expires in 7 days
2022-01-28 21:14:27 DEBUG    movie_list    clean.purge_reseed_seeds>purge.finished_seeds trying to match movie based off id imdb_id: tt9032400
2022-01-28 21:14:27 DEBUG    movie_list    clean.purge_reseed_seeds>purge.finished_seeds trying to match movie based of name: Eternals and year: 2021
2022-01-28 21:14:27 VERBOSE  limit_new     clean.purge_reseed_seeds>purge.finished_seeds Allowed Eternals.2021.BluRay.1080p.DTS-HD.MA5.1.x265.10bit-BeiTai ()
2022-01-28 21:14:27 DEBUG    limit_new     clean.purge_reseed_seeds>purge.finished_seeds Rejected: 0 Allowed: 1
2022-01-28 21:14:27 DEBUG    urlrewriter   clean.purge_reseed_seeds>purge.finished_seeds Checking 1 entries
2022-01-28 21:14:27 VERBOSE  details       clean.purge_reseed_seeds>purge.finished_seeds Summary - Accepted: 1 (Rejected: 2 Undecided: 0 Failed: 0)
2022-01-28 21:14:27 DEBUG    transmission  clean.purge_reseed_seeds>purge.finished_seeds Connecting to http://192.168.1.101:9091/transmission/
2022-01-28 21:14:27 DEBUG    transmission  clean.purge_reseed_seeds>purge.finished_seeds Successfully connected to transmission.
2022-01-28 21:14:27 DEBUG    transmission  clean.purge_reseed_seeds>purge.finished_seeds Found Eternals.2021.BluRay.1080p.DTS-HD.MA5.1.x265.10bit-BeiTai already loaded in transmission as Eternals.2021.BluRay.1080p.DTS-HD.MA5.1.x265.10bit-BeiTai
2022-01-28 21:14:27 INFO     transmission  clean.purge_reseed_seeds>purge.finished_seeds purged Eternals.2021.BluRay.1080p.DTS-HD.MA5.1.x265.10bit-BeiTai from transmission
2022-01-28 21:14:27 VERBOSE  list_match    clean.purge_reseed_seeds>purge.finished_seeds removing accepted entries from movie_list - symlinked-movies
2022-01-28 21:14:27 DEBUG    movie_list    clean.purge_reseed_seeds>purge.finished_seeds trying to match movie based off id imdb_id: tt9032400
2022-01-28 21:14:27 DEBUG    movie_list    clean.purge_reseed_seeds>purge.finished_seeds trying to match movie based of name: Eternals and year: 2021
2022-01-28 21:14:27 DEBUG    util.simple_persistence clean.purge_reseed_seeds>purge.finished_seeds Flushing simple persistence for task clean.purge_reseed_seeds>purge.finished_seeds to db.
2022-01-28 21:14:27 DEBUG    util.simple_persistence clean.purge_reseed_seeds>purge.finished_seeds Flushing simple persistence for task None to db.
2022-01-28 21:14:27 VERBOSE  from_task     clean.purge_reseed_seeds Finished running subtask `purge.finished_seeds`.
2022-01-28 21:14:27 VERBOSE  task          clean.purge_reseed_seeds ACCEPTED: `Eternals.2021.BluRay.1080p.DTS-HD.MA5.1.x265.10bit-BeiTai` by crossmatch plugin because intersects with Eternals.2021.BluRay.1080p.DTS-HD.MA5.1.x265.10bit-BeiTai on field(s) title
2022-01-28 21:14:27 VERBOSE  task          clean.purge_reseed_seeds ACCEPTED: `Eternals.2021.BluRay.1080p.DTS-HD.MA5.1.x265.10bit-BeiTai` by crossmatch plugin because intersects with Eternals.2021.BluRay.1080p.DTS-HD.MA5.1.x265.10bit-BeiTai on field(s) title
2022-01-28 21:14:27 VERBOSE  task          clean.purge_reseed_seeds ACCEPTED: `Eternals.2021.BluRay.1080p.DTS-HD.MA5.1.x265.10bit-BeiTai` by crossmatch plugin because intersects with Eternals.2021.BluRay.1080p.DTS-HD.MA5.1.x265.10bit-BeiTai on field(s) title
2022-01-28 21:14:27 DEBUG    urlrewriter   clean.purge_reseed_seeds Checking 3 entries
2022-01-28 21:14:27 VERBOSE  details       clean.purge_reseed_seeds Summary - Accepted: 3 (Rejected: 0 Undecided: 7 Failed: 0)
2022-01-28 21:14:27 WARNING  transmission  clean.purge_reseed_seeds Cannot purge Eternals.2021.BluRay.1080p.DTS-HD.MA5.1.x265.10bit-BeiTai because it is not loaded in transmission.
2022-01-28 21:14:27 WARNING  transmission  clean.purge_reseed_seeds Cannot purge Eternals.2021.BluRay.1080p.DTS-HD.MA5.1.x265.10bit-BeiTai because it is not loaded in transmission.
2022-01-28 21:14:27 WARNING  transmission  clean.purge_reseed_seeds Cannot purge Eternals.2021.BluRay.1080p.DTS-HD.MA5.1.x265.10bit-BeiTai because it is not loaded in transmission.
2022-01-28 21:14:27 DEBUG    util.simple_persistence clean.purge_reseed_seeds Flushing simple persistence for task clean.purge_reseed_seeds to db.
2022-01-28 21:14:27 DEBUG    util.simple_persistence clean.purge_reseed_seeds Flushing simple persistence for task None to db.
```
</details>

### Additional information:

- FlexGet version: 3.2.13
- Python version: 
- Installation method: docker/wiserain/flexget
- Using daemon (yes/no): yes
- OS and version:
- Link to crash log:

### PR DESCRIPTION
**Pull request by soloam**
### Motivation for changes:

More that one instance of transmission was not possible to use in plugin from_task

### Detailed changes:
- Save and restore preview transmission instance

### Addressed issues:
- Fixes #3331

### CODE REVIEW

**Comment by gazpachoking** [comment]
Hmm. This is why we don't like `__init__`s or storing things on `self` in the plugins. I wonder if there is a better way... Perhaps it gets stored on `task`? `task._transmission_client = Whatever()`?

**Comment by gazpachoking** [comment]
I suppose both of these solutions break if you do something like:
```yaml
inputs:
- from_transmission: hostA
- from_transmission: hostB
```

**Comment by gazpachoking** [comment]
Why are we even storing client instance like this? Shouldn't it just be created and closed in task_input (for from_transmission) and task_output (for regular transmission?)

**Comment by soloam** [comment]
I think that the main reason is to reuse the same connection if you use input and output or download (or all of them) in the same task. But now that I see it, like this you can't input from one server and output from another!

I also don't like to do this in the task start. One better way would be to start in the stages, and stor it in the task, and one each starge check if the config is the same, and if yes refuse, if not connect again.

**Comment by soloam** [comment]
I tried to re-trigger the checks, but something seams wrong! Locally was ok.

I changed the code to get get the client connection on every stage and store it to latter check if can be reused or make a new connection.

File: flexget/plugins/clients/transmission.py [hunk 1]
```diff
@@ -53,7 +54,7 @@ def prepare_config(self, config):
                 logger.error('netrc: {}, file: {}, line: {}', e.msg, e.filename, e.lineno)
         return config
 
-    def create_rpc_client(self, config):
+    def create_rpc_client(self, config) -> transmissionrpc.Client:
```
Comment (gazpachoking) [inline flexget/plugins/clients/transmission.py:57]:
```suggestion
    def create_rpc_client(self, config) -> 'transmissionrpc.Client':
```

This is probably the reason for the test failures.

**Comment by gazpachoking** [comment]
Thanks for this! Went with #3359 instead, as it seemed a bit simpler.

**Pull request closed without merging**